Address watchdogd service problem with generic script > 1s runtime #39

senthilnthangaraj opened the pull request:
This commit addresses a problem related to the watchdogd service. When trying to
activate a generic monitoring script (in this example it's /usr/sbin/my-script.sh) with a runtime exceeding one second,
it triggers an unintended system reboot.

Our configuration is as follows:

generic {
enabled = true
interval = 60
timeout = 20
warning = 1
critical = 10
monitor-script = "/usr/sbin/my-script.sh"
}

The error message we're encountering reads as below (even though my-script returns 0):

```
sv-isup my-service
watchdogd[661]: Monitor script PID 1056 still running after 20 sec
exit 0
watchdog: watchdog0: watchdog did not stop!

```

Upon further investigation, it was determined that the problem arises
from the fact that 'gs->script_runtime' is measured in milliseconds,
while 'gs->script_runtime_max' is maintained in seconds, as indicated by the source code here: [link to source code](https://github.com/troglobit/watchdogd/blob/master/src/generic.c#L68). This commit rectifies the issue.

## Unit test results
With the fix in this PR there is no failure in watchdog service and it works as expected, please see below the traces.

```
sv-isup my-service
exit 0

```

troglobit commented:
Please update the commit message to state what it does, and more importantly **why**.  Also mention how you have tested it to verify that it fixes the problem. Don't assume access to the GitHub issue tracker for this.

senthilnthangaraj commented:
> Please update the commit message to state what it does, and more importantly **why**. Also mention how you have tested it to verify that it fixes the problem. Don't assume access to the GitHub issue tracker for this.

@troglobit I have updated the commit message and added the UT results to the PR. Let me know if this looks good, thanks.
